src/main/java/bio/terra/cli/businessobject/Group.java

```diff
@@ -0,0 +1,202 @@
+    Map<String, Group> groups = new HashMap<>();
+    for (ManagedGroupMembershipEntry groupToPolicy : groupsToPolicies) {
+      String name = groupToPolicy.getGroupName();
+      String email = groupToPolicy.getGroupEmail();
+      Group group = groups.get(name);
+      if (group == null) {
+        group = new Group(name, email, new ArrayList<>());
+        groups.put(name, group);
+      }
+      GroupPolicy currentUserPolicy = GroupPolicy.fromSamPolicy(groupToPolicy.getRole());
+      group.currentUserPolicies.add(currentUserPolicy);
+    }
+    return groups;
+  }
+
+  /**
+   * Internal representation of a group member (i.e. someone who is a member of a SAM group). This
+   * is different from a regular {@link User} because they are never logged in. This is just a
+   * reference to another Terra user who has some group membership. This class is not part of the
+   * current context or state.
+   */
+  public static class Member {
+    private String email;
+    private List<GroupPolicy> policies;
+
+    private Member(String email, List<GroupPolicy> policies) {
+      this.email = email;
+      this.policies = policies;
+    }
+
+    public String getEmail() {
+      return email;
+    }
+
+    public List<GroupPolicy> getPolicies() {
+      return policies;
+    }
+  }
+
+  /**
+   * Add a member to a SAM group.
+   *
+   * @param email email of the user to add
+   * @param policy policy to assign the user
+   */
+  public Member addMember(String email, GroupPolicy policy) {
+    // call SAM to add a policy + email to the group
+    new SamService().addUserToGroup(name, policy, email);
+    logger.info("Added user to group: group={}, email={}, policy={}", name, email, policy);
+
+    // return a GroupMember = email + all policies (not just the one that was added here)
+    return getMember(email);
+  }
+
+  /**
+   * Remove a member from a SAM group.
+   *
+   * @param email email of the user to remove
+   * @param policy policy to remove from the user
+   */
+  public Member removeMember(String email, GroupPolicy policy) {
+    // call SAM to remove a policy + email from the group
+    new SamService().removeUserFromGroup(name, policy, email);
+    logger.info("Removed user from group: group={}, email={}, policy={}", name, email, policy);
+
+    // return a GroupMember = email + all policies (not just the one that was removed here)
+    return getMember(email);
+  }
+
+  /** Get the group member object. */
+  private Member getMember(String email) {
+    // lowercase the email so there is a consistent way of looking up the email address
+    // the email address casing in SAM may not match the case of what is provided by the user
+    return listMembersInMap().get(email.toLowerCase());
+  }
+
+  /** List the members of the group. */
+  public List<Member> listMembers() {
+    return listMembersInMap().values().stream().collect(Collectors.toList());
```

Comment:
alphabetize?

Reply:
why `listMembers` but `getCurrentUserPolicies`? I'd probably make it `getMembers()`.

Reply:
- No need to alphabetize here -- that's a display thing and is done in the command classes before outputing to the user. Here is just internal business logic, where we don't need a specific order.
- I think I used `list` because that's the name of the Sam endpoint I'm calling. But that shouldn't matter to callers. Changed to `getMembers`